de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/github/GithubImporterService.java

```diff
@@ -0,0 +1,265 @@
+import de.metas.issue.tracking.github.api.v3.model.Issue;
+import de.metas.issue.tracking.github.api.v3.model.Label;
+import de.metas.issue.tracking.github.api.v3.model.ResourceState;
+import de.metas.issue.tracking.github.api.v3.model.RetrieveIssuesRequest;
+import de.metas.issue.tracking.github.api.v3.service.GithubClient;
+import de.metas.logging.LogManager;
+import de.metas.organization.OrgId;
+import de.metas.serviceprovider.external.issuedetails.ExternalIssueDetail;
+import de.metas.serviceprovider.importer.ImportIssuesQueue;
+import de.metas.serviceprovider.importer.ImportService;
+import de.metas.serviceprovider.importer.info.ImportIssueInfo;
+import de.metas.serviceprovider.importer.info.ImportIssuesRequest;
+import de.metas.serviceprovider.milestone.Milestone;
+import de.metas.user.UserId;
+import de.metas.user.api.IUserDAO;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import de.metas.util.NumberUtils;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static de.metas.issue.tracking.github.api.v3.GitHubApiConstants.LabelType.BUDGET;
+import static de.metas.issue.tracking.github.api.v3.GitHubApiConstants.LabelType.ESTIMATION;
+import static de.metas.serviceprovider.external.issuedetails.ExternalIssueDetailType.LABEL;
+import static de.metas.serviceprovider.github.GithubImporterConstants.CHUNK_SIZE;
+import static de.metas.serviceprovider.github.GithubImporterConstants.HOUR_UOM_ID;
+import static de.metas.serviceprovider.importer.ImportConstants.IMPORT_LOG_MESSAGE_PREFIX;
+
+@Service
+public class GithubImporterService implements ImportService
+{
+	private static final Logger log = LogManager.getLogger(GithubImporterService.class);
+
+	private final ReentrantLock lock = new ReentrantLock();
+
+	private final ImportIssuesQueue importIssuesQueue;
+	private final GithubClient githubClient;
+	private final IUserDAO userDAO = Services.get(IUserDAO.class);
+
+	public GithubImporterService(final ImportIssuesQueue importIssuesQueue, final GithubClient githubClient)
+	{
+		this.importIssuesQueue = importIssuesQueue;
+		this.githubClient = githubClient;
+	}
+
+	public void start(@NonNull final ImmutableList<ImportIssuesRequest> requestList)
+	{
+		acquireLock();
+		try
+		{
+			final Stopwatch stopWatch = Stopwatch.createStarted();
+
+			requestList.forEach(request -> {
+				if (request.importByIds())
+				{
+					importIssuesById(request);
+				}
+				else
+				{
+					importIssues(request);
+				}
+			});
+			log.info(IMPORT_LOG_MESSAGE_PREFIX + " GithubImporterService#start() finished work in {} sec. ", stopWatch.stop().elapsed(TimeUnit.SECONDS));
```

Comment:
i think this is something for Loggable, too

Reply:
also u can use the stopWatch directly. It's default toString() is super user friendly. No need to convert it to seconds or anything else.

Reply:
another observation: when using logging APIs always avoid string concatenation